crates/plugins/mediator-coordination/src/pickup/handler.rs

```diff
@@ -0,0 +1,759 @@
+use crate::{
+    model::stateful::entity::{Connection, RoutedMessage},
+    pickup::{
+        constants::{MESSAGE_DELIVERY_3_0, PROBLEM_REPORT_2_0, STATUS_RESPONSE_3_0},
+        error::PickupError,
+        model::{
+            BodyDeliveryResponse, BodyLiveDeliveryChange, BodyStatusResponse, DeliveryResponse,
+            LiveDeliveryChange, StatusResponse,
+        },
+    },
+    web::{AppState, AppStateRepository},
+};
+use axum::response::{IntoResponse, Response};
+use didcomm::{Attachment, Message, MessageBuilder};
+use mongodb::bson::{doc, oid::ObjectId};
+use serde_json::Value;
+use std::{str::FromStr, sync::Arc};
+use uuid::Uuid;
+
+// Process pickup status request
+pub(crate) async fn handle_status_request(
+    state: Arc<AppState>,
+    message: Message,
+) -> Result<Message, Response> {
+    // Validate the return_route header
+    if message
+        .extra_headers
+        .get("return_route")
+        .and_then(Value::as_str)
```

Comment:
I think this is ok for now. The ticket responsible for splitting the protocols as separated crates will deal with that middleware. It will be exposed as a common utility to all the implementations that will need it.